tests/NClient.CodeGeneration/NClient.CodeGeneration.Interfaces.NSwag.Tests/NSwagGeneratorTest.cs

```diff
@@ -0,0 +1,138 @@
+                .Select(o => o.OperationId)
+                .ToList();
+            var availableOps = ops.Where(o => !notAvailableIds.Contains(o.OperationId)).ToList();
+            var opsCount = availableOps.Count;
+
+            var source = GenerateSourceCode(openApiSpec);
+
+            var type = Compile(source);
+            var methods = type.GetMethods();
+
+            methods.Should().HaveCount(opsCount);
+        }
+
+        private Type Compile(string source)
+        {
+            var syntaxTree = CSharpSyntaxTree.ParseText(source);
+
+            var assemblyName = Path.GetRandomFileName();
+            var ns = Assembly.Load("netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51");
+            #if !NETFRAMEWORK
+            var rt = Assembly.Load("System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
+            #endif
+            
+            MetadataReference[] references =
+            {
+                MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(System.Runtime.Serialization.EnumMemberAttribute).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Newtonsoft.Json.JsonConvert).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(System.ComponentModel.DataAnnotations.Validator).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Annotations.FacadeAttribute).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(Providers.Results.HttpResults.HttpResponse).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(GeneratedCodeAttribute).Assembly.Location),
+                MetadataReference.CreateFromFile(ns.Location),
+                #if !NETFRAMEWORK
+                MetadataReference.CreateFromFile(rt.Location)
+                #endif
+            };
+
+            var compilation = CSharpCompilation.Create(
+                assemblyName,
+                syntaxTrees: new[] { syntaxTree },
+                references: references,
+                options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+            using var ms = new MemoryStream();
+            // write IL code into memory
+            var result = compilation.Emit(ms);
+
+            if (!result.Success)
+            {
+                // handle exceptions
+                var failures = result.Diagnostics.Where(diagnostic =>
+                    diagnostic.IsWarningAsError ||
+                    diagnostic.Severity == DiagnosticSeverity.Error).ToList();
+
+                failures.Should().HaveCount(0, "Compilation should be successful");
+            }
+            ms.Seek(0, SeekOrigin.Begin);
+            var assembly = Assembly.Load(ms.ToArray());
+
+            // TODO: Check first
+            var type = assembly.GetTypes().Last(x => x.IsInterface);
```

Comment:
Now at least two interfaces are generated. In the specification, operations can be grouped using tags and then two interfaces will be created for each tag

Reply:
It is worth adding a [case with several tags](https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v2.0/json/uber.json) and also checking the names of the generated interfaces

Reply:
> Optional: when you execute the `dotnet ef` command, a funny message with help and version is displayed:
> 
> ```
> 
>                      _/\__
>                ---==/    \\
>          ___  ___   |.    \|\
>         | __|| __|  |  )   \\\
>         | _| | _|   \_/ |  //|\\
>         |___||_|       /   \\\/\\
> 
> Entity Framework Core .NET Command-line Tools 6.0.0
> 
> Usage: dotnet ef [options] [command]
> 
> Options:
>   --version        Show version information
>   -h|--help        Show help information
>   -v|--verbose     Show verbose output.
>   --no-color       Don't colorize output.
>   --prefix-output  Prefix output with level.
> 
> Commands:
>   database    Commands to manage the database.
>   dbcontext   Commands to manage DbContext types.
>   migrations  Commands to manage migrations.
> 
> Use "dotnet ef [command] --help" for more information about a command.
> ```
> 
> It would be great to do the same, if it's not difficult :)

Done

Reply:
Done